### ISSUE TITLE
Non-Code directive content incorrectly seen as CodeNode

### ISSUE THREAD

**Issue by weaverryan** [doctrine/rst-parser#90]
I'm not sure how to write a failing test case for this, so I'll describe the best I can.

Given both of these rst:

```rst
.. index::
   single: Installing and Setting up Symfony

Installing & Setting up the Symfony Framework
=============================================
```

```rst
.. tip::

    If you're having any problems running Symfony, your system may be missing
    some technical requirements. Use the :doc:`Symfony Requirements Checker </reference/requirements>`
    tool to make sure your system is set up.
```

The `single:` line is actually created as a `CodeNode` inside of `DocumentParser` - the same happens for the content inside of the `tip::`. But then, because we're inside a directive, it's not *actually* rendered as a `CodeNode`, it's passed to the process method of the current Directive:

https://github.com/doctrine/rst-parser/blob/master/lib/Parser/DocumentParser.php#L499-L509

This all happens because, at least for many directives, the `wantCode()` method returns true. And so, a `CodeNode` is "signaled" when parsing the directive: 

https://github.com/doctrine/rst-parser/blob/master/lib/Parser/DocumentParser.php#L372

The end result is that... everything renders fine. The directive's are passed a `CodeNode`... but they just call `$node->getValue()` and parse that however they need to.

The only problem (other than this is not a `CodeNode`... so it's technically wrong... but not really causing an issue) is that, if you listen to the `PostNodeCreateEvent::POST_NODE_CREATE` event, you will receive many `CodeNode` objects that are not actually code.

What's the purpose of the `Directive::wantCode()` method? Is it possibly `true` for too many directives (e.g. `SubDirective`, which is most directives)?

**Comment by jwage** [comment]
I honestly don't know. I haven't been able to figure that out yet. I've noticed it and knew it was wrong/weird and one day we'd have to fix it.

**Comment by weaverryan** [comment]
It's cool - we'll hopefully understand it a bit more and fix it sometime. It's work-aroundable :)

**Comment by wouterj** [comment]
@weaverryan I guess this is also fixed by your latest PR

**Comment by wouterj** [comment]
Closing as this should be fixed by #141 afaics
